spec/suites/layer/GeoJSONSpec.js

```diff
@@ -512,3 +512,364 @@ describe("L.LayerGroup#toGeoJSON", function () {
+			it("creates a Layer from a GeoJSON feature (type='" + geometry.type + "')", function () {
+				var layer = L.GeoJSON.geometryToLayer({
+					type: "Feature",
+					geometry: geometry
+				});
+				expect(layer).to.be.a(L.Layer);
+			});
+		});
+
+		[
+			point, line, polygon,
+			multiPoint, multiLine, multiPolygon,
+			geometryCollection
+		].forEach(function (geometry) {
+			it("creates a Layer from a GeoJSON geometry (type='" + geometry.type + "')", function () {
+				var layer = L.GeoJSON.geometryToLayer(geometry);
+				expect(layer).to.be.a(L.Layer);
+			});
+		});
+
+		it("throws an error if feature is an invalid GeoJSON object", function () {
+			expect(L.GeoJSON.geometryToLayer).withArgs({
+				type: "Feature",
+				geometry: {
+					type: "invalid",
+					coordinates: [0, 0]
+				}
+			}).to.throwError("Invalid GeoJSON object.");
+		});
+
+		it("returns nothing if feature does not have a geometry property", function () {
+			var ret = L.GeoJSON.geometryToLayer({type: "Feature"});
+			expect(ret).not.to.be.ok();
+		});
+
+		it("creates a Layer using pointToLayer option (Point)", function () {
+			var layer = L.GeoJSON.geometryToLayer({
+				type: "Feature",
+				geometry: point,
+				properties: {radius: 100}
+			}, {
+				pointToLayer: customPointToLayer
+			});
+			expect(layer).to.be.a(L.Circle);
+			expect(layer.options.radius).to.be(100);
+		});
+
+		it("creates a Layer using pointToLayer option (MultiPoint)", function () {
+			var layer = L.GeoJSON.geometryToLayer({
+				type: "Feature",
+				geometry: multiPoint,
+				properties: {radius: 100}
+			}, {
+				pointToLayer: customPointToLayer
+			});
+			layer.eachLayer(function (lyr) {
+				expect(lyr).to.be.a(L.Circle);
+				expect(lyr.options.radius).to.be(100);
+			});
+		});
+
+		it("creates a Layer using coordsToLatLng option (Point)", function () {
+			var layer = L.GeoJSON.geometryToLayer({
+				type: "Feature",
+				geometry: {
+					type: "Point",
+					coordinates: [1, 2, 3]
+				}
+			}, {
+				coordsToLatLng: customCoordstoLatLng
+			});
+			expect(layer.getLatLng()).to.eql({lat: 3, lng: 2, alt: 4});
+		});
+
+		it("creates a Layer using coordsToLatLng option (MultiPoint)", function () {
+			var layer = L.GeoJSON.geometryToLayer({
+				type: "Feature",
+				geometry: {
+					type: "MultiPoint",
+					coordinates: [
+						[1, 2, 3], [4, 5, 6]
+					]
+				}
+			}, {
+				coordsToLatLng: customCoordstoLatLng
+			});
+			expect(layer.getLayers().map(function (lyr) {
+				return lyr.getLatLng();
+			})).to.eql([
+				{lat: 3, lng: 2, alt: 4},
+				{lat: 6, lng: 5, alt: 7}
+			]);
+		});
+	});
+
+	describe("#coordsToLatLng", function () {
+		it("creates a LatLng object", function () {
+			var latLng = L.GeoJSON.coordsToLatLng([1, 2]);
+			var latLngWithAlt = L.GeoJSON.coordsToLatLng([3, 4, 5]);
+			expect(latLng).to.be.a(L.LatLng);
+			expect(latLngWithAlt).to.be.a(L.LatLng);
+		});
+
+		it("creates a LatLng object with given coordinates", function () {
```

Comment:
I'd suggest to unite this with prev. test

Reply:
The previous test should only test that the `L.GeoJSON.coordsToLatLng` function returns a L.LatLng object. This test only makes assertions against coordinates. For me these are two different cases and i learned unit testing to be more fine grained and to have in the best case only one assertion per test.

Reply:
> i learned unit testing to be more fine grained and to have in the best case only one assertion per test.

I suppose there can be different approaches for same thing.
For this case: I prefer to keep some balance between verbosity and maintainability.

Reply:
Done with 32ef194